plugins/com.google.cloud.tools.eclipse.appengine.localserver/src/com/google/cloud/tools/eclipse/appengine/localserver/server/LocalAppEngineServerBehaviour.java

```diff
@@ -315,17 +316,15 @@ public void onStart(Process process) {
     }
   }
 
-  private static final String URL_PATTERN = "http://[a-zA-Z0-9][a-zA-Z0-9-.]*[a-zA-Z0-9]:([0-9]+)";
-
   @VisibleForTesting
-  static int extractPortFromUrl(String line) {
+  static int extractPortFromServerUrlOutput(String line) {
     try {
-      Matcher match = Pattern.compile(URL_PATTERN).matcher(line);
-      if (match.find()) {
-        return Integer.parseInt(match.group(1 /* regex group for the port portion */));
+      int urlBegin = line.lastIndexOf("http://");
+      if (urlBegin != -1) {
+        return new URI(line.substring(urlBegin)).getPort();
       }
       return -1;
```

Comment:
We should javadoc `getServerPort()` and `getAdminPort()` with the possible responses.